app/src/main/java/org/android/go/sopt/presentation/auth/model/SignUpViewModel.kt

```diff
@@ -0,0 +1,76 @@
+package org.android.go.sopt.presentation.auth.model
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import org.android.go.sopt.data.dto.RequestSignUpDto
+import org.android.go.sopt.data.dto.ResponseSignUpDto
+import org.android.go.sopt.data.factory.ServicePool
+import org.android.go.sopt.util.addSourceList
+import org.android.go.sopt.util.enqueueUtil
+
+class SignUpViewModel : ViewModel() {
+
+    private val _signUpResult: MutableLiveData<ResponseSignUpDto> = MutableLiveData()
+    val signUpResult: LiveData<ResponseSignUpDto> = _signUpResult
+
+    val id: MutableLiveData<String> = MutableLiveData()
+    val pw: MutableLiveData<String> = MutableLiveData()
+    val name: MutableLiveData<String> = MutableLiveData()
+    val specialty: MutableLiveData<String> = MutableLiveData()
+
+    val isEnabledSignUpButton = MediatorLiveData<Boolean>().apply {
+        addSourceList(id, pw, name, specialty) {
+            checkSignUpValidation()
+        }
+    }
+
+    fun checkIdFormation(): Boolean {
+        return id.value?.matches(ID_RULE) ?: false && !id.value.isNullOrBlank()
```

Comment:
```suggestion
        return id.value?.matches(ID_RULE) && !id.value.isNullOrBlank()
```
`id.value?.matches(ID_RULE)` 자체가 패턴과 일치하는지 검사해주기 때문에 일치하면 true, 일치하지 않으면 false로 값을 알려줍니다 따라서 엘비스 연산자는 사용하지 않으셔도 됩니다

Reply:
오호 그러네요 댕장 수정하겠습니다~